### PR TITLE
fix: Fetch preview items also in the Get List sagas

### LINKED ISSUE #1893: The preview items of a list aren't loaded if the user only accesses the list
## 🎉 Description

If the user accesses the list via a link or through one of the toasts that appear, the items from the preview aren't loaded, making the share modal show a `ListCard` without items in the preview.

### PR DESCRIPTION
**Pull request by kevinszuchet**
Closes #1893 

Merge https://github.com/decentraland/marketplace-favorites-server/pull/186 first

### CODE REVIEW

**Review by flobarreto — APPROVED** [review]

File: webapp/src/modules/favorites/sagas.ts [hunk 1]
```diff
@@ -382,7 +392,19 @@ export function* favoritesSaga(getIdentity: () => AuthIdentity | undefined) {
         [favoritesAPI, 'getList'],
         id
       )
-      yield put(getListSuccess(list))
+
+      let { previewOfItemIds } = list
+
+      if (previewOfItemIds.length > 0) {
```
Comment (LautaroPetaccio) [inline webapp/src/modules/favorites/sagas.ts:398]: Loved this improvement, what if we move it to the `fetchPreviewItems` function so the `handleFetchListsRequest` can benefit from it as well?

File: webapp/src/modules/favorites/sagas.spec.ts [hunk 2]
```diff
@@ -1161,21 +1162,92 @@ describe('when handling the request for getting a list', () => {
...
+      items = [
+        {
+          id: 'anItemId',
+          name: 'anItemName'
+        } as Item
+      ]
+    })
+
+    describe('and non of the preview items are already in the state', () => {
+      it('should dispatch an action signaling the success of the handled action', () => {
```
Comment (LautaroPetaccio) [inline webapp/src/modules/favorites/sagas.spec.ts:1177]:
```suggestion
      it('should dispatch an action signaling the success of the handled action with the received list and items', () => {
```

File: webapp/src/modules/favorites/sagas.spec.ts [hunk 3]
```diff
@@ -1161,21 +1162,92 @@ describe('when handling the request for getting a list', () => {
...
+          ...items,
+          {
+            id: 'anotherItemId',
+            name: 'anotherItemName'
+          } as Item
+        ]
+        list = { ...list, previewOfItemIds: ['anItemId', 'anotherItemId'] }
+      })
+
+      it('should dispatch an action signaling the success of the handled action', () => {
```
Comment (LautaroPetaccio) [inline webapp/src/modules/favorites/sagas.spec.ts:1221]:
```suggestion
      it('should dispatch an action signaling the success of the handled action with the received list and the items that are not in the state', () => {
```

**Review by LautaroPetaccio — COMMENTED** [review]
Looks great 👍 I've left a comment to talk about

**Review by LautaroPetaccio — DISMISSED** [review]

**Review by LautaroPetaccio — APPROVED** [review]

**Pull request merged**